lib/mtl.js

```diff
@@ -1,118 +1,112 @@
-"use strict";
-var Promise = require('bluebird');
-var fs = require('fs-extra');
-var defined = require('cesium').defined;
+'use strict';
+var path = require('path');
+var readLines = require('./readLines');
 
-var fsReadFile = Promise.promisify(fs.readFile);
+module.exports = loadMtl;
 
-module.exports = {
-    getDefault : getDefault,
-    parse : parse
-};
-
-function createMaterial() {
-    return {
-        ambientColor : undefined,               // Ka
-        emissionColor : undefined,              // Ke
-        diffuseColor : undefined,               // Kd
-        specularColor : undefined,              // Ks
-        specularShininess : undefined,          // Ns
-        alpha : undefined,                      // d / Tr
-        ambientColorMap : undefined,            // map_Ka
-        emissionColorMap : undefined,           // map_Ke
-        diffuseColorMap : undefined,            // map_Kd
-        specularColorMap : undefined,           // map_Ks
-        specularShininessMap : undefined,       // map_Ns
-        normalMap : undefined,                  // map_Bump
-        alphaMap : undefined                    // map_d
-    };
+function Material() {
+    this.ambientColor = undefined;               // Ka
+    this.emissionColor = undefined;              // Ke
+    this.diffuseColor = undefined;               // Kd
+    this.specularColor = undefined;              // Ks
+    this.specularShininess = undefined;          // Ns
+    this.alpha = undefined;                      // d / Tr
+    this.ambientColorMap = undefined;            // map_Ka
+    this.emissionColorMap = undefined;           // map_Ke
+    this.diffuseColorMap = undefined;            // map_Kd
+    this.specularColorMap = undefined;           // map_Ks
+    this.specularShininessMap = undefined;       // map_Ns
+    this.normalMap = undefined;                  // map_Bump
+    this.alphaMap = undefined;                   // map_d
 }
 
-function getDefault() {
-    var material = createMaterial();
-    material.diffuseColor = [0.5, 0.5, 0.5, 1.0];
-    return material;
-}
+/**
+ * Parse an mtl file.
+ *
+ * @param {String} mtlPath Path to the mtl file.
+ * @returns {Promise} A promise resolving to the materials, or an empty object if the mtl file doesn't exist.
+ *
+ * @private
+ */
+function loadMtl(mtlPath) {
+    var material;
+    var values;
+    var value;
+    var materials = {};
+
+    function parseLine(line) {
+        line = line.trim();
```

Comment:
Is this trim necessary?  The regex can just ignore starting whitespace and you call trim a second time after getting the substring.  `trim` creates a new string every time and is not GC efficient so we should be careful with it.